flink-tests/src/test/java/org/apache/flink/test/recovery/ClusterEntrypointITCase.java

```diff
@@ -81,4 +87,45 @@ public void testWorkingDirectoryIsNotDeletedInCaseOfProcessFailure() throws Exce
             }
         }
     }
+
+    @Test
+    public void testNondeterministicWorkingDirectoryIsDeletedInCaseOfProcessFailure()
+            throws Exception {
+        final File workingDirBase = TEMPORARY_FOLDER.newFolder();
+
+        final Configuration configuration = new Configuration();
+        configuration.set(
+                ClusterOptions.PROCESS_WORKING_DIR_BASE, workingDirBase.getAbsolutePath());
+
+        final TestProcessBuilder.TestProcess jobManagerProcess =
+                new TestProcessBuilder(
+                                DispatcherProcess.DispatcherProcessEntryPoint.class.getName())
+                        .addConfigAsMainClassArgs(configuration)
+                        .start();
+
+        boolean success = false;
+        try {
+            CommonTestUtils.waitUntilCondition(
+                    () -> Files.list(workingDirBase.toPath()).findAny().isPresent(),
```

Comment:
The stream returned by Files.list needs to be closed.

Reply:
Will do.